File: app/src/main/java/com/microlabs/trallet/base/BaseApp.java
package com.microlabs.trallet.base;

import android.app.Application;

import com.crashlytics.android.Crashlytics;
import io.fabric.sdk.android.Fabric;
import io.realm.Realm;
import io.realm.RealmConfiguration;

/**
 * Base App Class for Realm Configuration
 */
public class BaseApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Fabric.with(this, new Crashlytics());
        Realm.init(this);
        RealmConfiguration realmConfiguration = new RealmConfiguration.Builder().build();
        Realm.setDefaultConfiguration(realmConfiguration);

        /*
        untuk menghapus seluruh isi database Realm
         */
        /*RealmConfiguration config = new RealmConfiguration.Builder()
                .deleteRealmIfMigrationNeeded()
                .build();
        Realm.deleteRealm(config);*/
    }
}

File: app/src/main/java/com/microlabs/trallet/base/BaseApp.kt
package com.microlabs.trallet.base

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import com.microlabs.trallet.BuildConfig
import timber.log.Timber

/**
 * Base App Class for Realm Configuration
 */
class BaseApp : Application() {

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
        val defaultNightMode = AppCompatDelegate.getDefaultNightMode()
//        Fabric.with(this, Crashlytics())
    }
}
